packit/upstream.py

```diff
@@ -195,3 +200,124 @@ def get_version(self) -> str:
             )
             return ups_ver
         return spec_ver
+
+    def get_current_version(self) -> str:
+        """
+        Get version of the project in current state (hint `git describe`)
+
+        :return: e.g. 0.1.1.dev86+ga17a559.d20190315 or 0.6.1.1.gce4d84e
+        """
+        ver = run_command(
+            self.package_config.current_version_command, output=True
+        ).strip()
+        logger.debug("version = %s", ver)
+        # FIXME: this might not work when users expect the dashes
+        #  but! RPM refuses dashes in version/release
+        ver = ver.replace("-", ".")
+        logger.debug("sanitized version = %s", ver)
+        return ver
+
+    def bump_spec(self, version: str = None, changelog_entry: str = None):
+        """
+        Run rpmdev-bumpspec on the upstream spec file: it enables
+        changing version and adding a changelog entry
+
+        :param version: new version which should be present in the spec
+        :param changelog_entry: new changelog entry (just the comment)
+        """
+        cmd = ["rpmdev-bumpspec"]
+        if version:
+            # 1.2.3-4 means, version = 1.2.3, release = 4
+            cmd += ["--new", version]
+        if changelog_entry:
+            cmd += ["--comment", changelog_entry]
+        cmd.append(self.specfile_path)
+        run_command(cmd)
+
+    def set_spec_version(self, version: str, changelog_entry: str):
+        """
+        Set version in spec and add a changelog_entry.
+
+        :param version: new version
+        :param changelog_entry: accompanying changelog entry
+        """
+        try:
+            # also this code adds 3 rpmbuild dirs into the upstream repo,
```

Comment:
@TomasTomecek may I ask you what is the problem here? We tried creating a new instance of SpecFile using our API and then calling methods set_version and update_changelog (from https://github.com/rebase-helper/rebase-helper/pull/598 ) and no such directories were created. Changelog comment has to be set in the constructor of a SpecFile instance.

We are aware that our API is not very consistent and is suited for your use case as of now. We are open to any changes that may help you.

Reply:
My bad, it's the invocation of rpmbuild which creates those dirs.

As for your API: I'd suggest to pass the changelog_entry as an argument of the method you've just implemented and not accept the changelog_entry as a constructor argument.